arch-secure-boot

```diff
@@ -75,39 +90,48 @@ case "$1" in
             /usr/lib/systemd/boot/efi/linuxx64.efi.stub "$NAME-unsigned.efi"
 
-        objcopy \
-            --add-section .osrel=/etc/os-release --change-section-vma .osrel=0x20000 \
-            --add-section .linux="/boot/vmlinuz-$KERNEL" --change-section-vma .linux=0x40000 \
-            --add-section .initrd="/boot/initramfs-$KERNEL-fallback.img" --change-section-vma .initrd=0x3000000 \
-            /usr/lib/systemd/boot/efi/linuxx64.efi.stub "$NAME-recovery-unsigned.efi"
-
-        objcopy \
-            --add-section .osrel=/etc/os-release --change-section-vma .osrel=0x20000 \
-            --add-section .linux=/boot/vmlinuz-linux-lts --change-section-vma .linux=0x40000 \
-            --add-section .initrd=/boot/initramfs-linux-lts-fallback.img --change-section-vma .initrd=0x3000000 \
-            /usr/lib/systemd/boot/efi/linuxx64.efi.stub "$NAME-recovery-lts-unsigned.efi"
+        for flavor in "${KERNEL#linux}" '-lts'; do
+            objcopy \
+                --add-section .osrel=/etc/os-release --change-section-vma .osrel=0x20000 \
+                --add-section .linux="/boot/vmlinuz-linux${flavor}" --change-section-vma .linux=0x40000 \
+                --add-section .initrd="/boot/initramfs-linux${flavor}-fallback.img" --change-section-vma .initrd=0x3000000 \
+                /usr/lib/systemd/boot/efi/linuxx64.efi.stub "${NAME%linux*}linux-recovery${flavor}-unsigned.efi"
+        done
 
-        for flavor in '' '-recovery' '-recovery-lts' '-efi-shell'; do
-            sbsign --key "$KEYSDIR/db.key" --cert "$KEYSDIR/db.crt" --output "${NAME}${flavor}.efi" "${NAME}${flavor}-unsigned.efi"
+        for flavor in "${KERNEL#linux}" "-recovery${KERNEL#linux}" '-recovery-lts' "${KERNEL#linux}-efi-shell"; do
+            sbsign --key "$KEYSDIR/db.key" --cert "$KEYSDIR/db.crt" --output "${NAME%linux*}linux${flavor}.efi" "${NAME%linux*}linux${flavor}-unsigned.efi"
         done
 
         sbsign --key "$KEYSDIR/db.key" --cert "$KEYSDIR/db.crt" /usr/lib/fwupd/efi/fwupdx64.efi
 
+        if [ -f "$ESP$SYSTEMD_BOOT" ]; then
+            sbsign --key "$KEYSDIR/db.key" --cert "$KEYSDIR/db.crt" --output "$ESP$SYSTEMD_SIGNED" "$ESP$SYSTEMD_BOOT"
+        fi
+
         mkdir -p "$ESP/$EFI"
         cp recovery.nsh "$ESP"
-        for flavor in '' '-recovery' '-recovery-lts' '-efi-shell'; do
-            cp "${NAME}${flavor}.efi" "$ESP/$EFI"
+        for flavor in "${KERNEL#linux}" "-recovery${KERNEL#linux}" '-recovery-lts' "${KERNEL#linux}-efi-shell"; do
+            cp "${NAME%linux*}linux${flavor}.efi" "$ESP/$EFI"
         done
         ;;
 
     add-efi)
         echo "Adding boot entries for EFI images..."
 
+        if ! findmnt /sys/firmware/efi/efivars >/dev/null; then
+            mount -o ro,nosuid,nodev,noexec,relatime efivarfs -t efivarfs /sys/firmware/efi/efivars
+        fi
+        [[ "$(findmnt -n -o OPTIONS -T /sys/firmware/efi/efivars)" =~ ^ro ]] && mount -o remount,rw efivarfs
```

Comment:
> Automatically mount efivarfs if it's not mounted and remount it with write enabled if it's mounted with read-only option.

I have never seen this before, interesting, thanks for sharing! I can't find any quick mentions of this, on internet or on arch wiki, could you please share some info or links, like how you achieve this and what is the primary benefit of doing so?

Also just curious, don't you want to unmount this afterwards? 🙂